src/analyzer/protocol/dns/DNS.cc

```diff
@@ -710,13 +708,79 @@ bool DNS_Interpreter::ParseRR_EDNS(DNS_MsgInfo* msg,
 			msg->BuildEDNS_Val()
 		);
 
-	// Currently EDNS supports the movement of type:data pairs
-	// in the RR_DATA section.  Here's where we should put together
-	// a corresponding mechanism.
-	if ( rdlength > 0 )
-		{ // deal with data
-		data += rdlength;
-		len -= rdlength;
+	// parse EDNS options
+	while ( len > 0 )
+		{
+		uint16_t option_code = ExtractShort(data, len);
+		int option_len = ExtractShort(data, len);
```

Comment:
Should there be any sanity checking for the option length in the switch below?

Reply:
In all honesty, [RFC6891](https://tools.ietf.org/html/rfc6891) indicates the fixed part of OPT RR has a length field and that is a *uint16_t* but in the variable portion, it only shows that the *OPTION-LENGTH* is 2 bytes and does not specify the type. It would make sense that its a *uint16_t* but it doesn't specify that.  I would be comfortable with the following courses of action.

1. *option_len* becomes a *uint16_t* to ensure it stays positive and is more inline with most length fields.
2. Inside the case statement, after matching the option_code, also validate *option_len* is large enough to fit the different types of addresses: 8 for IPv4, 20 for IPv6.

Is that what you were suggesting?

Reply:
Actually my numbers are off there with the 8 and 20 since the number of octets can be variable based on SOURCE PREFIX-LENGTH.

Reply:
@timwoj I updated the PR to include some validation of *option_len*.  Let me know if this is what you were looking for or if you think additional checking is required.

Reply:
Interesting enough, setting *option_len* to *uint16_t* required a typecast when calling ExtractShort.  On my Ubuntu 18.04 machine built with vagrant, no issues with the tests.  On the zeek test pipeline, it began parsing additional octets and therefore failed the tests.  I went back to using just an int and removed the typecast and now everything is passing once again.  My brain apparently needs more coffee.

Reply:
Yep, that's exactly what I was after.